### PR TITLE
[DISCO-2962] - Add cron job to gauge size of the region look up map

### PR DESCRIPTION
**Pull request by tiftran**
## References

JIRA: [DISCO-2962](https://mozilla-hub.atlassian.net/browse/DISCO-2962)

## Description
We have a look up map to help with retrieving the correct region for a given city, made changes so we can see the size of the map periodically. Debated whether we should log the contents of the map, but decided to hold off, until we look at the size?

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [ ] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2962]: https://mozilla-hub.atlassian.net/browse/DISCO-2962?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: merino/configs/default.toml [hunk 1]
```diff
@@ -218,6 +218,8 @@ connect_timeout_sec = 3.0
 # Global Redis settings. The weather provider optionally uses Redis to cache weather suggestions.
 # In the form of `redis://localhost:6379`.
 
+# MERINO_PROVIDERS__ACCUWEATHER__CRON_INTERVAL_SEC
+cron_interval_sec = 21600 # 6 hours
```
Comment (tiftran) [inline merino/configs/default.toml:222]: thoughts about this time interval?

File: merino/providers/weather/provider.py [hunk 2]
```diff
@@ -61,15 +67,30 @@ def __init__(
...
+            condition=self._should_fetch,
+            task=self._fetch_mapping,
+        )
+        self.cron_task = asyncio.create_task(cron_job())
 
     def hidden(self) -> bool:  # noqa: D102
         return False
 
+    def _should_fetch(self) -> bool:
+        return True
```
Comment (tiftran) [inline merino/providers/weather/provider.py:87]: I looked at the other providers and they periodcally check for last fetch time, in the event the previous fetch failed, but that's related to suggestion/content, while this is more of a good to know singular number.

File: merino/providers/weather/provider.py [hunk 3]
```diff
@@ -61,15 +67,30 @@ def __init__(
...
+
+    async def _fetch_mapping(self) -> None:
+        self.metrics_client.gauge(
+            name=f"providers.{self.name}.pathfinder.size", value=len(get_region_mapping())
+        )
+
```
Comment (Herraj) [inline merino/providers/weather/provider.py:93]: nit: Thoughts on renaming the function to `_fetch_mapping_size`?

File: merino/configs/default.toml (hunk 1 above)
Comment (Herraj) [reply merino/configs/default.toml:222]: I think that's good, should give us 4 measurements per day to start with.

**Review by Herraj — APPROVED** [review]
🗺️

File: merino/providers/weather/backends/accuweather/pathfinder.py [hunk 4]
```diff
@@ -87,6 +87,11 @@ def set_region_mapping(country: str, city: str, region: str | None):
         SUCCESSFUL_REGIONS_MAPPING[(country, city)] = region
 
 
+def get_region_mapping() -> dict:
```
Comment (ncloudioj) [inline merino/providers/weather/backends/accuweather/pathfinder.py:90]: The setter and the getter would behave very differently. Maybe call it `get_region_mapping_size()` and return the mapping size instead?

File: merino/providers/weather/provider.py [hunk 5]
```diff
@@ -61,15 +67,30 @@ def __init__(
...
 
     def hidden(self) -> bool:  # noqa: D102
         return False
 
+    def _should_fetch(self) -> bool:
+        return True
+
+    async def _fetch_mapping_size(self) -> None:
+        self.metrics_client.gauge(
+            name=f"providers.{self.name}.pathfinder.size", value=len(get_region_mapping())
```
Comment (ncloudioj) [inline merino/providers/weather/provider.py:91]:
```suggestion
            name=f"providers.{self.name}.pathfinder.mapping.size", value=len(get_region_mapping())
```

**Review by ncloudioj — APPROVED** [review]
r+wc.

**Pull request merged**